river/Seat.zig

```diff
@@ -318,30 +330,56 @@ pub fn handleMapping(
     const modes = &server.config.modes;
     for (modes.items[self.mode_id].mappings.items) |mapping| {
         if (std.meta.eql(modifiers, mapping.modifiers) and keysym == mapping.keysym and released == mapping.release) {
-            // Execute the bound command
-            const args = mapping.command_args;
-            var out: ?[]const u8 = null;
-            defer if (out) |s| util.gpa.free(s);
-            command.run(util.gpa, self, args, &out) catch |err| {
-                const failure_message = switch (err) {
-                    command.Error.Other => out.?,
-                    else => command.errToMsg(err),
-                };
-                std.log.scoped(.command).err("{s}: {s}", .{ args[0], failure_message });
-                return true;
-            };
-            if (out) |s| {
-                const stdout = std.io.getStdOut().writer();
-                stdout.print("{s}", .{s}) catch |err| {
-                    std.log.scoped(.command).err("{s}: write to stdout failed {}", .{ args[0], err });
+            if (mapping.repeat) {
+                self.mapping_repeat_command_args = mapping.command_args;
+                self.mapping_repeat_timer.timerUpdate(server.config.repeat_delay) catch {
+                    log.err("failed to update mapping repeat timer", .{});
                 };
             }
```

Comment:
Inconsistent UX: Tapping (pressing and releasing, not holding) a `-repeat` map will stop any actively repeating map, while tapping a normal bind will not.

I believe the right thing to do here is have *any* map stop currently repeating maps.

Reply:
Repeating mappings are stopped on any key press or release by a call to `clearRepeatingMapping` from `Keyboard.zig`. It seemed like mapped and unmapped keys should be treated the same as far as stopping repeats and the hook in Keyboard was the only way I saw to do that.